### PR TITLE
Bump go from 1.15 to 1.18

### PR DESCRIPTION
**Pull request by fengshunli**
### What changes are proposed in this pull request?

Please outline the changes and how this PR fixes the issue.

### Why are the changes needed?

 I think upgrading and splitting into sub-projects is essential

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

**Comment by ssz1997** [comment]
@fengshunli Thanks for your contribution. Are you planning on using CSI for your project?

**Comment by fengshunli** [comment]
If the project is deployed in the form of opertor, I think upgrading and splitting into sub-projects is essential

**Comment by fengshunli** [comment]
@elega @ssz1997 already approved

**Review by ssz1997 — CHANGES_REQUESTED** [review]
I'm ok with upgrading go version - we do need to do it for some other dependency. However is there a good reason to upgrade the alpine version?

File: integration/docker/Dockerfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ RUN cd /alluxio-csi && \
 #       - https://github.com/moby/moby/issues/6119
 # ADD with chown doesn't chown the files inside tarball
 #   See - https://github.com/moby/moby/issues/35525
-FROM alpine:3.10.2 AS alluxio-extractor
+FROM alpine:3.17 AS alluxio-extractor
```
Comment (Xenorith) [inline integration/docker/Dockerfile:31]: we cannot take in this alpine version change without extensive testing. i would advise to omit it from this PR

Comment (fengshunli) [reply integration/docker/Dockerfile:31]: version 1.18 is the most stable version that has undergone a lot of testing in many projects

Comment (Xenorith) [reply integration/docker/Dockerfile:31]: this is referring to the alpine version, not the golang version.

Comment (fengshunli) [reply integration/docker/Dockerfile:31]: alpine:3.17 is currently the most stable version with no cve vulnerabilities

**Comment by ssz1997** [comment]
Go in the new version of CSI has been updated to 1.19

**Pull request closed without merging**